eMarket/app/views.py

```diff
@@ -178,5 +180,33 @@ def order_detail(request, order_id):
 from django.contrib.auth import logout
 
 def logout_view(request):
-    logout(request)
-    return render(request, 'app/login.html')
+  logout(request)
+  return redirect('login')
+
+@login_required
+def seller_dashboard(request):
+    products = Product.objects.filter(seller=request.user).distinct()
+    orders = Order.objects.filter(cartproduct__product__seller=request.user).distinct()
```

Comment:
I can't seem to figure out how to link the orders and product to the seller. Could ya'll take a look at this and see how it could be done using the cartProduct model

Reply:
With the current structure of the database, the Order is not linked to the CartProduct model, it is linked to the Cart model. So, you should do cart__cartproduct__product__seller = request.user (long, i know)

Order → Cart (through cart field in Order)
Cart → CartProduct (through the reverse relationship on CartProduct)
CartProduct → Product (through the product field in CartProduct)
Product → Seller (through the seller field in Product)

Reply:
That solved the eroor